Task/1/schedule form

KuranovaPolina opened the pull request:
1 task done

KuranovaPolina committed:
Loading the project layout

KuranovaPolina committed:
Schedule.py added
created inline keyboards for schedule
Test code added to Schedule.py

KuranovaPolina committed:
Schedule form created
Inline buttons change "example" text in message

KuranovaPolina committed:
Adding test_schedule_data.py with data + changed cache_time in schedule.py

KuranovaPolina committed:
Added class Day + Inline keyboard "Переключить неделю" changes the week

KuranovaPolina committed:
Inline keyboards ">>" and "<<" scroll days and change week on boundary days

KuranovaPolina committed:
Added creating_formatted_text.py for formatting message text
Added schedule_text function for creating schedule message text (first test format)

KuranovaPolina committed:
schedule edited (new json format)

callmemars1 commented on .env.dist:
```diff
@@ -0,0 +1,14 @@
+DB_NAME=exampleDBName
+DB_HOST=127.0.0.1
+
+# need to rename file and put correctly data
```
Я так понимаю это файл с примером. Тогда зачем он в репозитории

callmemars1 commented on bot.py:
```diff
@@ -0,0 +1,61 @@
+import logging
+
+from aiogram import Bot, Dispatcher
+from aiogram.contrib.fsm_storage.redis import RedisStorage2
```
Зачем здесь редис?

LomasovEgor replied:
По приколу

LomasovEgor replied:
Я не знаю кто такой редис, но его ботяра как хранилище использует

LomasovEgor replied:
И вообще я не понимаю я это Сереже пишу в итоге или нет ахвпхавх. Сережа если ты это видешб мяукни в ответ

callmemars1 replied:
мяу, редис - это бд для кэширования ваще.

LomasovEgor replied:
пон

callmemars1 replied:
И все таки, редис зачем?

callmemars1 commented on tg_bot/handlers/echo.py:
```diff
@@ -0,0 +1,29 @@
+from aiogram import types, Dispatcher
```
Эхо в боте зачем

callmemars1 commented on tg_bot/handlers/schedule.py:
```diff
@@ -0,0 +1,82 @@
+    weekDay = datetime.isoweekday(datetime.today())
+
+
+async def schedule_schedule(message: Message):
```
Переименуй, непонятно что делает функция

LomasovEgor replied:
++

callmemars1 reviewed (changes requested):
Не совсем понимаю зачем на боте какие-то бд и админки. Много лишнего.
Часть папок содержит просто пустые файлы, их тоже надо убрать.
Надо еще будет протестировать бота, посмотреть как он выводит.
@LomasovEgor, проверь код, я основные моменты ток чекал.
По форматированию расписания:
Оставь один вариант названия недели (например только красная / синяя)
Номер пары можно не выводить

LomasovEgor commented on tg_bot/creating_formatted_text.py:
```diff
@@ -0,0 +1,52 @@
+             f"{week_name[week_type][0]} - " \
+             f"{week_name[week_type][1]} - " \
+             f"{week_name[week_type][2]} неделя</i>"
+    for i in test_schedule_data["lessons"]:
```
Переменную в for`е лучше так не называй, а то то потом в теле цикла непонятно чо куда.
Вот тут например вместо i лучше написать lesson.  for lesson in test_schedule_data['lessons']

LomasovEgor replied:
То что schedule лучше в timetable переименовать, чтобы у всех одинаково было, обсуждали уже

LomasovEgor commented on tg_bot/handlers/schedule.py:
```diff
@@ -0,0 +1,82 @@
+                                state="*")
+
+
+async def previous_day(call: CallbackQuery):
+    await call.answer(cache_time=1)
+
+    callback_data = call.data
+
+    logging.info(f"{callback_data=}")
+
+    if Day.weekDay == 1:
+        Day.weekType = 2 if Day.weekType == 1 else 1
+        Day.weekDay = 7
+    else:
+        Day.weekDay -= 1
+
+    await call.message.edit_text(text=schedule_text(Day.weekType, Day.weekDay), parse_mode="HTML")
+    await call.message.edit_reply_markup(reply_markup=days_scroll)
```
Функции  previous_day, next_day,  change_week в класс отдельный класс вынести в котором ты подобные функции прописываешь

KuranovaPolina replied:
Зачем?

LomasovEgor commented on tg_bot/handlers/schedule.py:
```diff
@@ -0,0 +1,82 @@
+    await call.message.edit_reply_markup(reply_markup=days_scroll)
+
+
+def register_pr_day(dp: Dispatcher):
+    dp.register_callback_query_handler(previous_day, text=["pr_day"])
```
Ну и функции регистрации  (register_pr_day, register_next_day, register_change_week) тоже в класс отдельный

KuranovaPolina replied:
И это тоже зачем?)

LomasovEgor replied:
Вот чтобы каждый раз когда новую функцию добавляешь тебе не пришлось потом в других файлах что-то дописывать. (вот как тут)
```python
from tg_bot.handlers.schedule import register_schedule, register_pr_day, register_next_day, register_change_week
```
Ты делаешь класс с этими функциями и импортируешь его.
```python
from tg_bot.handlers.schedule import *class_name*
```
и потом когда в бд загружаешь то тоже не вот так пишешь
```python
def register_all_handlers(dp):
    register_admin(dp)
    register_schedule(dp)
    register_pr_day(dp)
    register_next_day(dp)
    register_change_week(dp)
```
а создаешь метод в классе который регистрирует все. И когда будешь добавлять новый функционал или менять существующий тебе не нужно будет в разных файлах код дописывать, и все изменения будут в пределах одного класса.

KuranovaPolina replied:
До сих пор не вижу смысла в классах, но создала функцию register_full_timetable вызывающую все "регистрации"

LomasovEgor commented on tg_bot/creating_formatted_text.py:
```diff
@@ -0,0 +1,52 @@
+    if not lessons_finded:
+        result += f'\nВыходной'
+
+    return result
```
В питончике еще принято в конце всех файлов пустую строчку оставлять. Можешь просто периодически Ctrl+Alt+L нажимать и пайчарм все сам красиво сделает.

KuranovaPolina replied:
Спасибо, а то меня вечно эти подчеркивания в конце раздражают)

LomasovEgor commented on tg_bot/creating_formatted_text.py:
```diff
@@ -0,0 +1,52 @@
+}
+
+
+def schedule_text(week_type, week_day):
```
Функции лучше глаголами называть, а то у тебя тут будто это переменная

LomasovEgor replied:
Еще, по возможности, лучше писать тип возвращаемого значения у функций и тип данных у переменных.
def schedule_text(week_type: тип данных, week_day: тип данных) -> тип возвращаемого значения:
Это необязательно, но когда неочевидно лучше писать. Потом когда функцию вызываешь редактор будет показывать чо куда вводить и ругаться если другой тип дал.

LomasovEgor commented on bot.py:
```diff
@@ -0,0 +1,61 @@
+from tg_bot.config import load_config
+from tg_bot.filters.admin import AdminFilter
+from tg_bot.handlers.admin import register_admin
+from tg_bot.handlers.schedule import register_schedule, register_pr_day, register_next_day, register_change_week
```
Я там ниже писал, что нужно функции в класс занести. Вот тут просто его импортируешь его и кайфуешь. И не придется каждый раз тут новую функцию импортировать.

LomasovEgor commented on bot.py:
```diff
@@ -0,0 +1,61 @@
+    dp.filters_factory.bind(AdminFilter)
+
+
+def register_all_handlers(dp):
+    register_admin(dp)
+    register_schedule(dp)
+    register_pr_day(dp)
+    register_next_day(dp)
+    register_change_week(dp)
```
Ну и тут соответственно через класс просто все делаешь.

KuranovaPolina committed:
renaming: schedule -> timetable and the others

KuranovaPolina committed:
some files added to .gitignore (deleted)

KuranovaPolina committed:
some files added to .gitignore (deleted) - 2

KuranovaPolina committed:
.gitignore (deleted) edited

KuranovaPolina committed:
All unnecessary files added to .gitignore (deleted)

KuranovaPolina committed:
Unnecessary code from bot.py are deleted

KuranovaPolina committed:
Added new package test_data, text_format
file test_timetable_data.py moved to test_data
file timetable_text.py moved to text_format

KuranovaPolina committed:
.gitignore edited

KuranovaPolina committed:
create_timetable -> display_timetable

KuranovaPolina committed:
timetable style edited

KuranovaPolina committed:
added input, output types and types of variables

KuranovaPolina committed:
timetable_text.py code style edited

KuranovaPolina committed:
Added register_full_timetable function in timetable.py
It is called from bot.py instead of all registers

callmemars1 commented on requirements.txt:
```diff
@@ -0,0 +1,3 @@
+aiogram>=2.0,<3.0
+aioredis==2.0
```
Здесь тоже редис проставлен

KuranovaPolina committed:
Database DELETED!!!

callmemars1 commented on tg_bot/config.py:
```diff
@@ -0,0 +1,50 @@
+
+
+@dataclass
+class DbConfig:
```
бд

KuranovaPolina replied:
хм... самое интересное, что я убрала от сюда уже бд...

callmemars1 commented on bot.py:
```diff
@@ -0,0 +1,44 @@
+    config = load_config(".env")
+
+    bot = Bot(token=config.tg_bot.token, parse_mode="HTML")
+    storage = RedisStorage2() if config.tg_bot.use_redis else MemoryStorage()
```
бд

KuranovaPolina replied:
И от сюда

callmemars1 commented on tg_bot/config.py:
```diff
@@ -0,0 +1,50 @@
+class TgBot:
+    token: str
+    admin_ids: list[int]
+    use_redis: bool
```
редис

callmemars1 commented on tg_bot/config.py:
```diff
@@ -0,0 +1,50 @@
+        tg_bot=TgBot(
+            token=env.str("BOT_TOKEN"),
+            admin_ids=list(map(int, env.list("ADMINS"))),
+            use_redis=env.bool("USE_REDIS"),
```
редис

callmemars1 commented on tg_bot/config.py:
```diff
@@ -0,0 +1,50 @@
+            admin_ids=list(map(int, env.list("ADMINS"))),
+            use_redis=env.bool("USE_REDIS"),
+        ),
+        db=DbConfig(
```
бд

callmemars1 commented on bot.py:
```diff
@@ -0,0 +1,44 @@
+
+    bot = Bot(token=config.tg_bot.token, parse_mode="HTML")
+    storage = RedisStorage2() if config.tg_bot.use_redis else MemoryStorage()
+    dp = Dispatcher(bot, storage=storage)
```
Из конструктора storage можно убрать

KuranovaPolina committed:
storage deleted from bot.py

callmemars1 commented on bot.py:
```diff
@@ -0,0 +1,44 @@
+    try:
+        await dp.start_polling()
+    finally:
+        await dp.storage.close()
```
Тк нет бд, то и нет необходимости ее закрывать

KuranovaPolina committed:
storage deleted from bot.py 2

callmemars1 commented on tg_bot/handlers/timetable.py:
```diff
@@ -0,0 +1,96 @@
+from tg_bot.text_format.timetable_text import format_timetable_text
+
+
+class Day:
```
### Это нельзя делать статиками!!!
Что будет если два пользователя захотят одновременно получить расписание на разные дни / типы недель

KuranovaPolina replied:
Это получается для каждого пользователя надо создавать отдельный объект класса Day?

callmemars1 replied:
да

callmemars1 commented on tg_bot/config.py:
```diff
@@ -0,0 +1,28 @@
+    return Config(
+        tg_bot=TgBot(
+            token=env.str("BOT_TOKEN"),
+            admin_ids=list(map(int, env.list("ADMINS"))),
```
Админов мы убрали
Бтв то, что у тебя там заигнорено, перестанет работать

KuranovaPolina replied:
хм.. а если мы будем дальше развивать бота и нам надо будет что-то только для админов проверять?

callmemars1 replied:
Тогда мы это добавим, но пока этого в планах нет

callmemars1 commented on tg_bot/handlers/timetable.py:
```diff
@@ -0,0 +1,96 @@
+    await call.message.edit_reply_markup(reply_markup=days_scroll)
+
+
+def register_pr_day(dp: Dispatcher):
```
Замени pr на previous или prev для читаемости

KuranovaPolina committed:
Changed pr to previous
Finally deleted storage

KuranovaPolina committed:
Changed pr to previous
Finally deleted storage

KuranovaPolina committed:
Changed pr to previous
Finally deleted storage

KuranovaPolina committed:
Merge remote-tracking branch 'origin/task/1/schedule-form' into task/1/schedule-form

KuranovaPolina committed:
Changed pr to previous 2

LomasovEgor commented on tg_bot/creating_formatted_text.py:
```diff
@@ -0,0 +1,52 @@
+}
+
+
+def schedule_text(week_type, week_day):
```
Функции глаголами называть надо

LomasovEgor commented on tg_bot/creating_formatted_text.py:
```diff
@@ -0,0 +1,52 @@
+             f"{week_name[week_type][0]} - " \
+             f"{week_name[week_type][1]} - " \
+             f"{week_name[week_type][2]} неделя</i>"
+    for i in test_schedule_data["lessons"]:
```
Переменную в for`е лучше так не называй, а то то потом в теле цикла непонятно чо куда.
Вот тут например вместо i лучше написать lesson. for lesson in test_schedule_data['lessons']

KuranovaPolina committed:
Added users.py for saving info about users
All user have personal data info

KuranovaPolina committed:
Unnecessary code deleted

callmemars1 reviewed (commented):
Ладно, пока пускай будет это, фиксить будем в следующих тасках
@LomasovEgor, посмотри и апрувни если тебя всё устроит
